tests/ztimer_xsec/main.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2021 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief    high level ztimer test application
+ *
+ * @author   Karl Fessel <karl.fessel@ovgu.de>
+ *
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "ztimer.h"
+#include "mutex.h"
+
+#ifndef LOG_LEVEL
+#define LOG_LEVEL LOG_WARNING
+#endif
+
+#include "log.h"
+
+typedef struct named_lock {
+    char *name;
+    mutex_t mut;
+} named_lock_t;
+
+void release(void *arg);
+
+static named_lock_t sec_lock = { .name = "SEC", .mut = MUTEX_INIT_LOCKED };
+static named_lock_t msec_lock = { .name = "MSEC", .mut = MUTEX_INIT_LOCKED };
+static named_lock_t usec_lock = { .name = "USEC", .mut = MUTEX_INIT_LOCKED };
+
+static ztimer_t sec_tim = { .callback = release, .arg =  &sec_lock };
+static ztimer_t msec_tim = { .callback = release, .arg = &msec_lock };
+static ztimer_t usec_tim = { .callback = release, .arg = &usec_lock };
+void release(void *arg)
+{
+    named_lock_t *e = arg;
+
+    LOG_INFO("time: %" PRIu32 "\n", ztimer_now(ZTIMER_USEC));
+
+    printf("unlocking %s\n", e->name);
```

Comment:
stdio in ISR causes the ISR stack to overflow. IMO it is best to avoid stdio in ISR

Reply:
should i do two puts instead?

Reply:
i went with one